packages/services/src/view-service/transaction-planner/fees.ts

```diff
@@ -0,0 +1,34 @@
+import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+
+// Attempts to extract a fee token from the assets used in the actions of the planner request
+// Priority in descending order
+export const extractAltFee = (request: TransactionPlannerRequest): AssetId => {
+  const outputAsset = request.outputs.map(o => o.value?.assetId).find(Boolean);
+  if (outputAsset) return outputAsset;
+
+  const spendAsset = request.spends.map(o => o.value?.assetId).find(Boolean);
+  if (spendAsset) return spendAsset;
+
+  const swapAsset = request.swaps.map(assetIn => assetIn.value?.assetId).find(Boolean);
+  if (swapAsset) return swapAsset;
+
+  const auctionScheduleAsset = request.dutchAuctionScheduleActions
+    .map(a => a.description?.outputId)
```

Comment:
The output of an auction is the asset the user is trying to buy, so the user may not have any of it yet. I think you want the input asset ID, right?

Reply:
nice catch I switched those in my head, this should be `input?.assetId`